crates/astria-sequencer/src/grpc/optimistic.rs

```diff
@@ -0,0 +1,152 @@
+use std::{
+    pin::Pin,
+    sync::Arc,
+};
+
+use astria_core::{
+    generated::sequencerblock::v1::optimisticblock::v1alpha1::{
+        optimistic_block_service_server::OptimisticBlockService,
+        GetBlockCommitmentStreamRequest,
+        GetBlockCommitmentStreamResponse,
+        GetOptimisticBlockStreamRequest,
+        GetOptimisticBlockStreamResponse,
+    },
+    primitive::v1::RollupId,
+    Protobuf,
+};
+use tonic::{
+    codegen::tokio_stream::{
+        wrappers::ReceiverStream,
+        Stream,
+    },
+    Request,
+    Response,
+    Status,
+};
+use tracing::{
+    debug,
+    info,
+};
+
+use crate::app::OptimisticBlockChannels;
+
+pub(crate) struct OptimisticBlockServer {
+    optimistic_block_channels: OptimisticBlockChannels,
+}
+
+impl OptimisticBlockServer {
+    pub(crate) fn new(optimistic_block_channels: OptimisticBlockChannels) -> Self {
+        Self {
+            optimistic_block_channels,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl OptimisticBlockService for OptimisticBlockServer {
+    type GetBlockCommitmentStreamStream =
```

Comment:
this is a weird name, I believe they just suffix the rpc method name with `Stream`. I think we can avoid the word stream in our rpc methods?